cpp/src/arrow/util/visibility.h

```diff
@@ -20,6 +20,7 @@
 
 #if defined(_WIN32) || defined(__CYGWIN__)
 #if defined(_MSC_VER)
+#pragma warning(push)
```

Comment:
It this needed?
If we need this, do we also need `warning(pop)`?

Reply:
removing this